pkg/syncer/cnsoperator/controller/cnsregistervolume/util.go

```diff
@@ -391,3 +393,109 @@ func isPVCBound(ctx context.Context, client clientset.Interface, claim *v1.Persi
 	return false, fmt.Errorf("persistentVolumeClaim %s in namespace %s not in phase %s within %d seconds",
 		pvcName, ns, v1.ClaimBound, timeoutSeconds)
 }
+
+// updateBackingDiskAnnotation checks if the backing disk annotation on the PVC matches
+// the backing disk type mentioned on CnsRegisterVolume instance.
+func updateBackingDiskAnnotation(ctx context.Context, k8sClient clientset.Interface,
+	instance *cnsregistervolumev1alpha1.CnsRegisterVolume,
+	pvc *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error) {
+	log := logger.GetLogger(ctx)
+	log.Infof("Checking BackingType for PVC %s.", pvc.Name)
+
+	if pvc.Annotations == nil {
+		pvc.Annotations = make(map[string]string)
+	}
+
+	backingDiskType, backingDiskAnnExists := pvc.Annotations[common.AnnKeyBackingDiskType]
+	if backingDiskAnnExists {
+		if backingDiskType == instance.Spec.BackingType {
+			log.Infof("BackingType for PVC %s is up to date. Skip updating.", pvc.Name)
```

Comment:
We do not support updating CNS Register Volume Spec after status is done.

If we need to change BackingType after PVC/PV are created and CNS Register Volume CR status is done, we need to unregister and re-register volume again.